pylint/checkers/base.py

```diff
@@ -298,28 +298,40 @@ def _get_break_loop_node(break_node):
 
 def _loop_exits_early(loop):
     """
-    Returns true if a loop may end with a break statement.
+    Returns true if a loop may end with a break or return statement.
 
     Args:
         loop (astroid.For, astroid.While): the loop node inspected.
 
     Returns:
-        bool: True if the loop may end with a break statement, False otherwise.
+        bool: True if the loop may end with a break or return statement, False otherwise.
     """
     loop_nodes = (nodes.For, nodes.While)
```

Comment:
Personally I think this whole function could do with a rewrite. Using a `for` loop with premature returns (and potentially breaks) would improve readability and could probably remove unnecessary assignments such as `definition_nodes` and `break_nodes`.